### ISSUE TITLE
Slim 3.0

### ISSUE THREAD

**Issue by gricey432** [entomb/slim-json-api#27]
With the Slim 3.0 release candidate out, I've tried to get this to work but I'm getting
```
Fatal error: Class 'Slim\Middleware' not found in /var/www//vendor/entomb/slim-json-api/jsonAPI/JsonApiMiddleware.php on line 23
```

There have been some large changes to how middleware works, http://www.slimframework.com/docs/concepts/middleware.html

Will this package be updated to work with 3.0? I love it and want to keep using it

**Comment by entomb** [comment]
hi @gricey432, yeah i'll try to keep it up to date with slim3, thanks for the head's up!

**Comment by gricey432** [comment]
@entomb Awesome, thanks

**Comment by kriang69** [comment]
@gricey432  How to solve it?
`Fatal error: Class 'Slim\View' not found in C:\xampp\htdocs\slim\test\vendor\entomb\slim-json-api\jsonAPI\JsonApiView.php on line 24`

`Fatal error: Class 'Slim\Middleware' not found in C:\xampp\htdocs\slim\test\vendor\entomb\slim-json-api\jsonAPI\JsonApiMiddleware.php on line 23`

**Comment by entomb** [comment]
@kriang69 how are you including the lib? it should be autoloaded if you use composer, if not, you need to include the two files manually.

**Comment by jfastnacht** [comment]
Is anyone working on this?

**Comment by biganfa** [comment]
just use `/Slim/Http/Response.php:withJson(...)` method

**Comment by entomb** [comment]
@jfastnacht our friend @biganfa is correct , Slim3 has native json response support with `withJson()` and therefore you don't need this library anymore.

**Comment by lukaszb1** [comment]
blablanla `<?php eho 'test'; ?>`

**Comment by rinogo** [comment]
@entomb Yes, but one advantage of this middleware is that it automatically catches exceptions and returns them as JSON.  Is there an easy equivalent implementation of this in Slim 3.0 without a bunch of repetitive boilerplate code?

**Comment by rinogo** [comment]
For others who end up here with the same question, this is the solution I ended up using:
```
// Set up error handler
$c = $app->getContainer();
$c["errorHandler"] = function($c) {
	return function ($request, $response, $exception) use ($c) {
		return $c['response']->withJson(array("error" => $exception->getMessage()), 500);
	};
};
```
